src/librustc_trans/trans/context.rs

```diff
@@ -977,11 +942,62 @@ fn declare_intrinsic(ccx: &CrateContext, key: &str) -> Option<ValueRef> {
                 ccx.intrinsics().borrow_mut().insert($name, f.clone());
                 return Some(f);
             }
+        );
+        ($name:expr, $cname:ident ($($arg:expr),*) -> $ret:expr) => (
+            if !is_like_pnacl {
+                // The `if key == $name` is already in ifn!
+                ifn!($name, fn($($arg),*) -> $ret);
+            } else if key == $name {
+                let f = declare::declare_cfn(ccx, stringify!($cname),
+                                             Type::func(&[$($arg),*], &$ret),
+                                             ccx.tcx().mk_nil());
+                ccx.intrinsics().borrow_mut().insert($name, f.clone());
+                return Some(f);
+            }
         )
     }
 
     compatible_ifn!("llvm.assume", noop(llvmcompat_assume(i1) -> void), 6);
 
+    // These must be redirected to `libm` here so PNaCl's linker will link in
+    // these functions from `libm`.
+    compatible_ifn!("llvm.copysign.f32", copysignf(t_f32, t_f32) -> t_f32);
```

Comment:
They can't because the linker might omit linking to the libm versions (PNaCl is completely static, so the linker can make such decisions vs regular platform where libc/libm is shared), so it *must* be done here.